lib/search-field/index.jsx

```diff
@@ -36,21 +42,37 @@ export class SearchField extends Component {
     }
   }
 
-  interceptEsc = event => {
-    if (KEY_ESC === event.keyCode) {
-      if (this.state.query === '') {
-        this.inputField.blur();
-      }
-      this.clearQuery();
+  doSearch = query => {
+    this.setState({ query, searchSelected: true });
+    this.debouncedSearch(query);
```

Comment:
let's remove the `query` from the debounce to prevent the parallel-universes effect when we have multiple calls to `doSearch` in a row. granted, we shouldn't have the same problems as with the note content (which could be updated by remote calls) but this keeps a single source of data instead of two

`debouncedSearch()` also has to change but not in a big way

```js
debouncedSearch = debounce(() => this.props.onSearch(this.state.query), SEARCH_DELAY);
```

Reply:
is `debounce` the appropriate idiom for search? if we keep on typing then it won't search until we stop. a `throttle` might be more appropriate so that we can see search results as we continue to type but it will still make sure we don't send a new search on every keystroke

`debounce` will prevent the search from interfering with our typing but won't update until we have finished typing
`throttle` will allow a slow search to interfere with our typing but provide us partial results as we type